src/types/index.ts

```diff
@@ -1,48 +1,5 @@
-import { ec } from 'elliptic';
-import {
-  MintRequest,
-  GetSignableTransferRequestV1,
-  GetSignableTransferRequest,
-  Configuration as APIConfiguration,
-} from '../api';
-import { GetSignableBurnRequest } from '../workflows/types';
-import { Signer as L1Signer } from '@ethersproject/abstract-signer';
-
-export { L1Signer };
-
-export interface L2Signer {
-  signMessage(message: string): Promise<string>;
-  getAddress(): string | Promise<string>;
-}
-
-export interface StarkWallet {
-  path: string;
-  starkPublicKey: string;
-  starkKeyPair: ec.KeyPair;
-}
-
-export interface WalletConnection {
-  l1Signer: L1Signer;
-  l2Signer: L2Signer;
-}
-
-export interface L1Configuration {
-  coreContractAddress: string;
-  registrationContractAddress: string;
-  chainID: number;
-}
-
-export interface ImmutableXConfiguration {
-  apiConfiguration: APIConfiguration;
-  l1Configuration: L1Configuration;
-}
-
-export type UnsignedMintRequest = Omit<MintRequest, 'auth_signature'>;
-export type UnsignedTransferRequest = GetSignableTransferRequestV1;
-export type UnsignedBatchNftTransferRequest = GetSignableTransferRequest;
-export type UnsignedBurnRequest = GetSignableBurnRequest;
-
-export * from './deposit';
-export * from './withdrawal';
-export * from './token';
-export * from './signable-withdrawal';
+export * from './generic';
```

Comment:
file name suggestion: `publicInterfaces` instead of `generic`. thoughts @matthewmuscat @zatlite ?

Reply:
@Kastet How about we split the generic file into `signerInterfaces` and `configInterfaces`?

Reply:
or config interfaces can go into `config.ts` and this file can be renamed to `signers.ts`?

Reply:
[69dd24a](https://github.com/immutable/imx-core-sdk/pull/185/commits/69dd24a1f887bea007bc7a3990422774e93afbdf)